acceptance/steps/k8s.go

```diff
@@ -63,6 +68,45 @@ func assertVariableValue(ctx context.Context, t framework.TestingT, variableName
 	require.Equal(t, previousValue, currentValue)
 }
 
+func assertVariableValueIncremented(ctx context.Context, t framework.TestingT, variableName, jsonPath, groupVersionKind, resourceName string) {
+	currentValue := execJSONPath(ctx, t, jsonPath, groupVersionKind, resourceName)
+	previousValue := ctx.Value(recordedVariable(variableName))
+
+	if reflect.TypeOf(previousValue) != reflect.TypeOf(currentValue) {
+		t.Fatalf("unmatched types: %T, %T", previousValue, currentValue)
+		return
+	}
+
+	// check if we're dealing with integer types
+	// NOTE: this verbose switch statement is painful
+	// but there's not a great way of incrementing
+	// a number otherwise via reflection
```

Comment:
Not asking for a change but I think you could get away with up casting this to an int64 and adding 1? `reflect.ValueOf(value).Int() + 1`